Vagrantfile

```diff
@@ -6,55 +6,39 @@ VAGRANTFILE_API_VERSION = "2"
 
 Vagrant.require_version ">= 1.7.0"
 
+require_relative './provisioning/key_authorization'
+
+
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   config.vm.box = "ubuntu/trusty64"
   cchq_proxy_port = ENV.fetch("VAGRANT_CCHQ_PROXY_PORT", 8080)
   config.ssh.insert_key = false
-
-  config.vm.define "app1" do |app1|
-    app1.vm.hostname = "app1"
-    app1.vm.network "private_network", ip: "192.168.33.15"
-    app1.vm.provider "virtualbox" do |v|
-      v.memory = 768
-      v.cpus = 1
+  authorize_key_for_root config, '~/.ssh/id_dsa.pub', '~/.ssh/id_rsa.pub'
+
+  {
+    'app1'    => '192.168.33.15',
+    'db1'   => '192.168.33.16',
+    'proxy1' => '192.168.33.17',
+  }.each do |short_name, ip|
+    config.vm.define short_name do |host|
+      host.vm.network 'private_network', ip: ip
+      host.vm.hostname = "#{short_name}.hq.dev"
+      host.vm.provision "shell", path: "provisioning/nodes.sh"
+      host.vm.provider "virtualbox" do |v|
```

Comment:
I recently learned that vagrant supports docker. Thoughts on using docker instead of virtualbox one day?

Reply:
Sounds interesting